### PR TITLE
@itwin/eslint-plugin 4.0.0-dev.34

### PR DESCRIPTION
**Pull request by pmconne**
Includes https://github.com/iTwin/eslint-plugin/pull/15 to err on unbound static method calls.

### CODE REVIEW

File: common/config/rush/pnpm-lock.yaml [hunk 1]
```diff
@@ -5747,11 +5733,11 @@ packages:
       '@itwin/core-quantity': 3.6.0 - 3.6.1
       '@itwin/ecschema-metadata': 3.6.0 - 3.6.1
     dependencies:
-      '@itwin/core-backend': link:../../core/backend
-      '@itwin/core-bentley': link:../../core/bentley
-      '@itwin/core-common': link:../../core/common
-      '@itwin/core-geometry': link:../../core/geometry
-      '@itwin/ecschema-metadata': link:../../core/ecschema-metadata
+      '@itwin/core-backend': link:..\..\core\backend
```
Comment (pmconne) [inline common/config/rush/pnpm-lock.yaml:5736]: @arun do OS-specific path separators surprise/concern you?

**Comment by pmconne** [comment]
@paulius-valiunas should I be required to manually install @itwin/eslint-plugin's peer dependencies?
See e.g. extract-api CI output.
`rush install` locally produces, for each package:
```
../../utils/workspace-editor
└─┬ @itwin/eslint-plugin
  ├── ✕ missing peer @typescript-eslint/eslint-plugin@~5.55.0
  ├── ✕ missing peer eslint-plugin-deprecation@^1.3.3
  ├── ✕ missing peer eslint-plugin-jam3@^0.2.3
  ├── ✕ missing peer eslint-plugin-jsdoc@^35.1.3
  ├── ✕ missing peer eslint-plugin-jsx-a11y@^6.4.1
  ├── ✕ missing peer eslint-plugin-react@^7.32.2
  └── ✕ missing peer eslint-plugin-react-hooks@^4.6.0
Peer dependencies that should be installed:
  @typescript-eslint/eslint-plugin@~5.55.0  eslint-plugin-jam3@^0.2.3                 eslint-plugin-jsx-a11y@^6.4.1             eslint-plugin-react@^7.32.2
  eslint-plugin-deprecation@^1.3.3          eslint-plugin-jsdoc@^35.1.3               eslint-plugin-react-hooks@^4.6.0
```

File: common/config/rush/pnpm-lock.yaml (hunk 1 above)
Comment (aruniverse) [reply common/config/rush/pnpm-lock.yaml:5736]: this is new, never noticed it before. idk if we should be concerned about it yet, will look into it

**Comment by paulius-valiunas** [comment]
> @paulius-valiunas should I be required to manually install @itwin/eslint-plugin's peer dependencies? See e.g. extract-api CI output. `rush install` locally produces, for each package:
> 
> ```
> ../../utils/workspace-editor
> └─┬ @itwin/eslint-plugin
>   ├── ✕ missing peer @typescript-eslint/eslint-plugin@~5.55.0
>   ├── ✕ missing peer eslint-plugin-deprecation@^1.3.3
>   ├── ✕ missing peer eslint-plugin-jam3@^0.2.3
>   ├── ✕ missing peer eslint-plugin-jsdoc@^35.1.3
>   ├── ✕ missing peer eslint-plugin-jsx-a11y@^6.4.1
>   ├── ✕ missing peer eslint-plugin-react@^7.32.2
>   └── ✕ missing peer eslint-plugin-react-hooks@^4.6.0
> Peer dependencies that should be installed:
>   @typescript-eslint/eslint-plugin@~5.55.0  eslint-plugin-jam3@^0.2.3                 eslint-plugin-jsx-a11y@^6.4.1             eslint-plugin-react@^7.32.2
>   eslint-plugin-deprecation@^1.3.3          eslint-plugin-jsdoc@^35.1.3               eslint-plugin-react-hooks@^4.6.0
> ```

modern package managers allow installing peer dependencies automatically. You can add `auto-install-peers=true` to the `.npmrc` to enable that in rush/pnpm: https://github.com/microsoft/rushstack/issues/3356#issuecomment-1290988220

**Comment by pmconne** [comment]
> You can add auto-install-peers=true to the .npmrc to enable that in rush/pnpm

@aruniverse had some (unspecified) reservations about doing that.

I tried it just now, and `rush install`, `rush install --purge`, `rush update`, and `rush update --full` still failed to install the eslint plugin's  peer dependencies.

**Comment by paulius-valiunas** [comment]
> I tried it just now, and `rush install`, `rush install --purge`, `rush update`, and `rush update --full` still failed to install the eslint plugin's peer dependencies.
Perfect timing, I just ran into the same issue in GPB... Looks like I forgot to remove one reference to `eslint-plugin-prefer-arrow` in the config when I removed it from dependencies. I added it manually to GPB's dependencies and it started working - can you try that too, just to see if there are any other issues?

**Comment by paulius-valiunas** [comment]
https://github.com/iTwin/eslint-plugin/pull/17

**Comment by aruniverse** [comment]
> > You can add auto-install-peers=true to the .npmrc to enable that in rush/pnpm
> 
> 
> 
> @aruniverse had some (unspecified) reservations about doing that.
> 
> 
> 
> I tried it just now, and `rush install`, `rush install --purge`, `rush update`, and `rush update --full` still failed to install the eslint plugin's  peer dependencies.

@paulius-valiunas we need to upgrade to pnpm v7 for that. We're still using pnom V6 under the hood. 

I know the pnpm creator was cautious of adding that feature too. My main concern is people are going to rely on that feature as a crutch and not understand how dependencies work. It's also a security concerns imo. Eg some pkg lists a btc miner as a peer dep, now it'll get auto installed. Might be unbased and unlikely to happen but it's an extreme edge case.

**Comment by paulius-valiunas** [comment]
> > > You can add auto-install-peers=true to the .npmrc to enable that in rush/pnpm
> > 
> > 
> > @aruniverse had some (unspecified) reservations about doing that.
> > I tried it just now, and `rush install`, `rush install --purge`, `rush update`, and `rush update --full` still failed to install the eslint plugin's  peer dependencies.
> 
> @paulius-valiunas we need to upgrade to pnpm v7 for that. We're still using pnom V6 under the hood.
> 
> I know the pnpm creator was cautious of adding that feature too. My main concern is people are going to rely on that feature as a crutch and not understand how dependencies work. It's also a security concerns imo. Eg some pkg lists a btc miner as a peer dep, now it'll get auto installed. Might be unbased and unlikely to happen but it's an extreme edge case.

I don't get the security angle. How is it different from said dependency installing a bitcoin miner as a "normal" dependency?

And I can't believe core is still using pnpm v6, I'm already migrating things to v8.

**Comment by aruniverse** [comment]
> And I can't believe core is still using pnpm v6, I'm already migrating things to v8.

We will need this to go through to take advantage of the new auto-install feature. Will try to get someone to take a look at this asap. https://github.com/iTwin/itwinjs-core/pull/4996

**Pull request closed without merging**